### PR TITLE
Add Short.io extension

### PR DESCRIPTION
**Pull request by koinzhang**
## Description

- Shorten links using your own domain via Short.io.

**Get your secret API key [here](https://app.short.io/settings/integrations/api-key)**:

- Click "Create API key".
- Add a Secret key.

## Screencast

Metadata-1

Metadata-2

Metadata-3

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/short-io/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Short.io Changelog
+
+## [Initial Version] - {PR_MERGE_DATE}
+
+- Shorten links using your own domain via Short.io.
```
Comment (danielsequeira) [inline extensions/short-io/CHANGELOG.md:5]: Maybe you could also mention the other commands, at least the ability to search for shorten links

File: extensions/short-io/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Short.io
+
+Shorten links using your own domain via Short.io.
+
+**Tips**:
+
+- The first time you use 𝐒𝐡𝐨𝐫𝐭𝐞𝐧 𝐋𝐢𝐧𝐤 and 𝐒𝐞𝐚𝐫𝐜𝐡 𝐋𝐢𝐧𝐤𝐬 commands, they will use the first domain of your domains list.
+- You can set default domain for 𝐒𝐡𝐨𝐫𝐭𝐞𝐧 𝐋𝐢𝐧𝐤 and 𝐒𝐞𝐚𝐫𝐜𝐡 𝐋𝐢𝐧𝐤𝐬 commands via the 𝐒𝐡𝐨𝐫𝐭𝐞𝐧 𝐋𝐢𝐧𝐤 𝐖𝐢𝐭𝐡 𝐃𝐨𝐦𝐚𝐢𝐧 command (action: Set Default Domain, shortcut: Cmd+Enter).
+
+**Get your secret API key [here](https://app.short.io/settings/integrations/api-key)**:
```
Comment (danielsequeira) [inline extensions/short-io/README.md:10]: I tried with a public API key and got a 401, probably worth pointing out that users should create a private API key

File: extensions/short-io/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Short.io
+
+Shorten links using your own domain via Short.io.
+
+**Tips**:
+
+- The first time you use 𝐒𝐡𝐨𝐫𝐭𝐞𝐧 𝐋𝐢𝐧𝐤 and 𝐒𝐞𝐚𝐫𝐜𝐡 𝐋𝐢𝐧𝐤𝐬 commands, they will use the first domain of your domains list.
+- You can set default domain for 𝐒𝐡𝐨𝐫𝐭𝐞𝐧 𝐋𝐢𝐧𝐤 and 𝐒𝐞𝐚𝐫𝐜𝐡 𝐋𝐢𝐧𝐤𝐬 commands via the 𝐒𝐡𝐨𝐫𝐭𝐞𝐧 𝐋𝐢𝐧𝐤 𝐖𝐢𝐭𝐡 𝐃𝐨𝐦𝐚𝐢𝐧 command (action: Set Default Domain, shortcut: Cmd+Enter).
```
Comment (danielsequeira) [inline extensions/short-io/README.md:8]: Sorry for nitpicking, but could you keep all in the same font? I see the commands in a different one like: 𝐒𝐞𝐚𝐫𝐜𝐡 𝐋𝐢𝐧𝐤𝐬. If you want to highlight you can use bold or code instead

File: extensions/short-io/package.json [hunk 4]
```diff
@@ -0,0 +1,78 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "short-io",
+  "title": "Short.io",
+  "description": "Shorten links using your own domain via Short.io.",
+  "icon": "short-io-icon.png",
```
Comment (danielsequeira) [inline extensions/short-io/package.json:6]: Shouldn't their main icon be this one instead: https://play-lh.googleusercontent.com/BdiGYSn4Dfb7QBhDH0qzmLwMeWUKvIT76RsEh8-B9e30tkh79X82lQxE-X_ExcdldA4t

At least that is the one I see in their website

File: extensions/short-io/package.json [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      "description": "Shorten links using your own domain and set default domain of other commands.",
+      "mode": "view"
+    }
+  ],
+  "preferences": [
+    {
+      "name": "apiKey",
+      "type": "password",
+      "required": true,
+      "title": "API Secret key",
```
Comment (danielsequeira) [inline extensions/short-io/package.json:44]:
```suggestion
      "title": "API Secret Key",
```

File: extensions/short-io/package.json [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+      "description": "Short.io API secret key.",
+      "default": ""
```
Comment (danielsequeira) [inline extensions/short-io/package.json:46]: default is not a required property, you can remove it

**Review by danielsequeira — COMMENTED** [review]
Thanks for another contribution @koinzhang, all looks good, I just commented in a few details.

File: extensions/short-io/CHANGELOG.md (hunk 1 above)
Comment (koinzhang) [reply extensions/short-io/CHANGELOG.md:5]: Thanks for the suggestion, I changed the description to:
`Shorten and search links with your own domain via Short.io.`
How do you think?

File: extensions/short-io/README.md (hunk 3 above)
Comment (koinzhang) [reply extensions/short-io/README.md:8]: Thanks for the suggestion, I will use Blod to bold them.

File: extensions/short-io/CHANGELOG.md (hunk 1 above)
Comment (danielsequeira) [reply extensions/short-io/CHANGELOG.md:5]: sounds good 👍

File: extensions/short-io/README.md (hunk 2 above)
Comment (koinzhang) [reply extensions/short-io/README.md:10]: I haven't tested the public key yet, I'll test it first, and if there is indeed a problem, I'll modify the README to tell the user to request the private key

Comment (koinzhang) [reply extensions/short-io/README.md:10]: I added tips to the README: 
- Do not turn on the _Public key_ switch, this extension does not support public key.

**Review by danielsequeira — APPROVED** [review]

**Pull request merged**